### PR TITLE
Fix #6417: users with group permission should be able to create groups via API

### LINKED ISSUE #6417: Unable to create groups by non-admin thru API
Even though I can create groups in GitLab itself (as I am granted with privilege to create groups) I cannot find a way how to force API to create group with current user as owner of the group and thus chance to add projects and members through API.

In other words: POST /groups should not be admin only, as not only admin has right to create group.

But following change (mad with respect to group_controller) doesn't cut it:

lib/api/groups.rb - line 43

```ruby
post do
  required_attributes! [:name, :path]

  attrs = attributes_for_keys [:name, :path]
  @group = Group.new(attrs)

  if @group.save
    @group.add_owner(current_user)
    present @group, with: Entities::Group
  else
    not_found!
  end
end
```

This creates group with name and path as intended, but current_user in not added to this group and whole presented result is a nil.

### PR DESCRIPTION
**Pull request by jirutka**
This is updated and slightly modified patch from #7529 that fixes #6417.

### CODE REVIEW

**Comment by NARKOZ** [comment]
LGTM

**Comment by deric** [comment]
+1

File: spec/requests/api/groups_spec.rb [hunk 1]
```diff
@@ -3,8 +3,9 @@
 describe API::API, api: true  do
   include ApiHelpers
 
-  let(:user1) { create(:user) }
+  let(:user1) { create(:user, can_create_group: false) }
   let(:user2) { create(:user) }
+  let(:user3) { create(:user) }
```
Comment (jvanbaarsen) [inline spec/requests/api/groups_spec.rb]: Is it needed to create a user3? Can't we just use user2?

Comment (jirutka) [inline spec/requests/api/groups_spec.rb]: Yes, it’s needed. Read the code, you’ll see.

Comment (jvanbaarsen) [inline spec/requests/api/groups_spec.rb]: Well, I see you use it. But in the diff I don;t see a reason you can't use user2. can you please explain in a little bit more wording why it is needed? I have about 100+ PR's that I need to review. I can't go and source dive all the time. Thats the reason I ask those questions. 
Its not my intention to insult you or anything like that. If you feel that way, I'm sorry.

I need to be sure that a PR is good before we can merge it. If I have the feeling it is not good I'm forced to close it without merging. That wouldn't help you, nor the GitLab project.

Comment (jirutka) [inline spec/requests/api/groups_spec.rb]: We need a user that is not member of a group. Both user1 and user2 are set to be members of on of the groups.
I’ve send this PR one month ago, I really don’t remember all the details now…

Comment (jvanbaarsen) [inline spec/requests/api/groups_spec.rb]: Ok, thanks for explaining :) :tada:

**Comment by randx** [comment]
Thanks

**Pull request merged**